classes/Address.php

```diff
@@ -440,14 +440,23 @@ public static function getCountryAndState($id_address)
      *
      * @return bool The address exists
      */
-    public static function addressExists($id_address)
+    public static function addressExists($id_address, $refreshCache = false): bool
     {
-        return (bool) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue(
-            'SELECT `id_address`
-            FROM ' . _DB_PREFIX_ . 'address a
-            WHERE a.`id_address` = ' . (int) $id_address,
-            false
-        );
+        if ($id_address == 0) {
```

Comment:
This will patch the specific bug but I think the actual bug is why we're calling this method over and over with id = 0

In reality the same would be true with `null`, `''` and any value lower than zero.

Reply:
👍🏾 

Maybe you can do

```
0 === (int) $id_address
```

Reply:
I agree with @eternoendless that this fixes the problem locally and the symptom but we should worry about the source. I guess there are places where the address is not set yet (like maybe the delivery address in the cart) so the default value is 0 but we still check if it exists which doesn't make much sense

There are two approaches the:
- modify each code using this exists and precheck if the address is set before calling exists, although it might be a big modification
- the present solution, which is a shortcut, might be enough but the problem is also that it might hide unexpected side effects We assume "no address" is the same as "address not found", but in some cases, both should be handled differently

Reply:
I agree with @eternoendless  and @jolelievre, this is just fixing the last step before we query the database. The main issue is why we check 30 times in the class `Cart` in the first time. The calculation in the cart is very messy and does the same logic over and over again.